modules/flowable-ui-idm/flowable-ui-idm-rest/src/main/java/org/flowable/ui/idm/rest/app/IdmSSOResource.java

```diff
@@ -0,0 +1,181 @@
+
+        if (urlBase.startsWith("http://") && "https".equals(request.getHeader(HttpHeaders.X_FORWARDED_PROTO))) {
+            urlBase = urlBase.replace("http://", "https://");
+        }
+
+        if (ssoHandler.isActive()) {
+            response.getWriter().write(
+                ssoHandler.getExternalUrl(urlBase)
+            );
+            response.flushBuffer();
+        }
+    }
+
+    @RequestMapping(value = "/sso/return", method = RequestMethod.GET)
+    public void onSsoReturnGet(HttpServletRequest request, HttpServletResponse response) {
+        handleSSOReturn(request, response, null);
+    }
+
+    @RequestMapping(value = "/sso/return", method = RequestMethod.POST)
+    public void onSsoReturn(@RequestBody MultiValueMap<String, String> body, HttpServletRequest request, HttpServletResponse response) {
+        handleSSOReturn(request, response, body);
+    }
+
+    private void handleSSOReturn(HttpServletRequest request, HttpServletResponse response, MultiValueMap<String, String> body) {
+        SSOUserInfo ssoUserInfo = ssoHandler.handleSsoReturn(request, response, body);
+
+        if (ssoUserInfo != null) {
+            try {
+                UserInformation userInfo = userService.getUserInformation(ssoUserInfo.getId());
+                loginWithSso(request, response, userInfo.getUser());
+
+            } catch (NotFoundException e) {
+                // userInfo not found: create new account
+                createSsoAccount(request, response, ssoUserInfo);
+
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            response.setStatus(HttpStatus.UNAUTHORIZED.value());
+        }
+    }
+
+    private static Logger LOGGER = LoggerFactory.getLogger(IdmSSOResource.class);
+
+    private void loginWithSso(HttpServletRequest request, HttpServletResponse response, User user) throws IOException {
+        String redirectTo = request.getRequestURL().toString().replace("/app/sso/return", "");
+
+        if (request.getCookies() != null) {
+            Optional<Cookie> optional = Arrays.stream(request.getCookies())
+                .filter(cookie -> "redirectUrl".equals(cookie.getName()))
+                .findFirst();
+
+            if (optional.isPresent()) {
+                redirectTo = URLDecoder.decode(optional.get().getValue());
+            }
+        }
+
+        response.setStatus(HttpServletResponse.SC_MOVED_PERMANENTLY);
+        response.setHeader("Location", redirectTo);
+        rememberMeServices.loginSuccess(request, response, new SSOAuthentication(user));
+        response.getOutputStream().flush();
+    }
+
+    private void createSsoAccount(HttpServletRequest request, HttpServletResponse response, SSOUserInfo userInfo) {
+        userService.createNewUser(
+            userInfo.getId(),
+            userInfo.getFirstName(),
+            userInfo.getLastName(),
+            userInfo.getEmail(),
+            UUID.randomUUID().toString()
```

Comment:
The password is created so that no-one can just enter a username without password and get logged in. But the created user also gets forced to login through the SSO this way.